src/shared/reducers/message.js

```diff
@@ -16,10 +16,22 @@ import {
   FETCH_FAILURE,
 } from "../actions/constants";
 
-export const addErrorToState = (state, { error }) => ({
-  ...state,
-  message: error,
-});
+export const addErrorToState = (state, { error }) => {
+  let message = error;
+
+  if (message instanceof Error) {
+    ({ message } = error);
+  }
+
+  if (typeof message !== "string") {
+    throw new Error("invalid error message. A string is expected here.");
```

Comment:
This is not really accurate since `message` can be either an error or a string. Yet, when the error object is used, it is necessarily a string. I would do that instead:

```
export const addErrorToState = (state, { error }) => {
  let message;
  if (error instanceof Error) {
    message = error.message;
  } else if (typeof error === "string") {
    message = error;
  } else {
    throw new Error("addErrorToState requires either an Error or a string");
  }
};
```

Reply:
Done